server/src/main/java/org/elasticsearch/ingest/Processor.java

```diff
@@ -94,13 +95,20 @@ Processor create(Map<String, Processor.Factory> processorFactories, String tag,
          * instances that have run prior to in ingest.
          */
         public final ThreadContext threadContext;
+    
+        /**
+         * Provides a thread pool
+         */
+        // TODO: do we really want to expose ThreadPool here? Or a BiFunction<Long, Runnable, ScheduledFuture<?>> to just handle scheduling?
```

Comment:
is this something you want to discuss?
I see no issue with leaving the Threadpool available to processors

Reply:
It is a shame that we have to expose anything at all so let us at least go with the minimum necessary.

Reply:
So I had doubt about directly exposing TP to all processors, because this will send to wrong message to processor implementors, there should be no need to fork a thread. I will expose the minimum necessary.